Fix/controllerprofile

the6th opened the pull request:
@tarukosu 

fix small issues on Controller example.

1. Typo Congroller to Controller ,and related profiles settings.

2. Improve the method to find MagicLeap Controller on Start.

I found this patch brings to work fine on Unity2019.1.x + unitypackage 0.21.0  also.

the6th committed:
fix typo: Congroller to Controller

the6th committed:
fix:[try repair] problem on MagicLeapControllerSample.scene

the6th committed:
improve logic for finding MLController

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,6 +99,38 @@ private void StartMLInput()
+                    MLInput.OnControllerConnected += HandleOnControllerConnected;
+                    MLInput.OnControllerDisconnected += HandleOnControllerDisconnected;
+                    MLInput.OnControllerButtonUp += HandleOnButtonUp;
+                    MLInput.OnControllerButtonDown += HandleOnButtonDown;
```
136-139 と重複していますが、ここで設定しておく必要があるのでしょうか？

the6th replied:
これは消し忘れです。消します！

the6th replied:
https://github.com/HoloLabInc/MRTKExtensionForMagicLeap/commit/84ed83b0faf9c2c0f58d1706aad3e8b8a537b864
で一通り調整しました。

再レビューお願いします！

tarukosu replied:
MLInput が他のスクリプトでスタートされてしまう場合を考えると、
元の箇所のほうがいい気がするのですがいかがでしょうか？

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,6 +99,38 @@ private void StartMLInput()
+                    //in that case the callback is never called.
+                    //try to find connected controllers.
+
+                    var con = MLInput.GetController(MLInput.Hand.Right);
```
con -> controller にしていただけますか？

tarukosu replied:
var controller = MLInput.GetController(MLInput.Hand.Right) ?? MLInput.GetController(MLInput.Hand.Left)
if(controller != null){
  HandleOnControllerConnected(controller.Id);
}

tarukosu replied:
こんな感じでどうでしょう？

the6th replied:
そうですね。これにします。

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,6 +99,38 @@ private void StartMLInput()
+
+                    }
+
+                    mlInputStarted = true;
```
MLInputConfiguration config = new MLInputConfiguration(MLInputConfiguration.DEFAULT_TRIGGER_DOWN_THRESHOLD,

の前の行に入れてもいいでしょうか？

MLInput.Start に失敗するときに、StartMLInput が毎ループ呼ばれるのが気になります

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -43,7 +43,7 @@ public class MagicLeapDeviceManager : BaseInputDeviceManager, IMixedRealityCapab
             {
                 StartMLInput();
-                mlInputStarted = true;
+                //mlInputStarted = true;
```
もとのままでいいかと思ったのですが、 mlInputStarted = true; はStartMLInput に入れたほうが良いですか？

the6th replied:
失敗したときに、復帰できた方が良いかと思ったのですが、
レアケースなので、元に戻しますね。

もう古いバージョンですが、
2019.1.x + 0.21.0のUnityPackageは、
MLInput.StartがMonoBehaviourのAwakeで呼ばれることが前提だったので、
コントローラの検知が、MRTKのこのタイミングだと、遅かったんですよね。
そのあたりを調整していた時の名残で、ここになってるだけなので。。

the6th committed:
fix pointed out in PR

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,12 +99,28 @@ private void StartMLInput()
+                    MLInput.OnControllerButtonUp += HandleOnButtonUp;
+                    MLInput.OnControllerButtonDown += HandleOnButtonDown;
+
+                    //when using SDK 0.21.0+Unity2019.1.x
```
```suggestion
                    // When using SDK 0.21.0+Unity2019.1.x
```

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,12 +99,28 @@ private void StartMLInput()
+                    MLInput.OnControllerButtonDown += HandleOnButtonDown;
+
+                    //when using SDK 0.21.0+Unity2019.1.x
+                    //Controller might be already connected at this frame.
```
```suggestion
                    // controllers might be already connected at this frame.
```

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,12 +99,28 @@ private void StartMLInput()
+
+                    //when using SDK 0.21.0+Unity2019.1.x
+                    //Controller might be already connected at this frame.
+                    //in that case the callback is never called.
```
```suggestion
                    // In that case the callback is never called.
```

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,12 +99,28 @@ private void StartMLInput()
+                    //when using SDK 0.21.0+Unity2019.1.x
+                    //Controller might be already connected at this frame.
+                    //in that case the callback is never called.
+                    //try to find connected controllers.
```
```suggestion
                    // That's why try to find connected controllers.
```

tarukosu commented on Assets/MixedRealityToolkit.ThirdParty/MagicLeapInput/Scripts/MagicLeapDeviceManager.cs:
```diff
@@ -99,12 +99,28 @@ private void StartMLInput()
+                    {
+                        HandleOnControllerConnected(controller.Id);
+                    }
+
```
```suggestion
```

tarukosu committed:
Apply suggestions from code review

the6th committed:
Apply suggestions from code review(callback)

https://github.com/HoloLabInc/MRTKExtensionForMagicLeap/pull/3/files#r352998312
